File: flutter-packages/mobile/lib/constants/time.dart
String timeAgo(int millis) {
  if (millis == 0) return '';
  final dt = DateTime.fromMillisecondsSinceEpoch(millis);
  final diff = DateTime.now().difference(dt);

  if (diff.inDays >= 7) {
    final w = (diff.inDays / 7).floor();
    return '${w}w ago';
  } else if (diff.inDays >= 1) {
    return '${diff.inDays}d ago';
  } else if (diff.inHours >= 1) {
    return '${diff.inHours}h ago';
  } else if (diff.inMinutes >= 1) {
    return '${diff.inMinutes}m ago';
  } else {
    return 'now';
  }
}

File: app/ratel/src/common/utils/time.rs
pub fn get_now_timestamp() -> i64 {
    chrono::Utc::now().timestamp()
}

pub fn get_now_timestamp_millis() -> i64 {
    chrono::Utc::now().timestamp_millis()
}

pub fn get_now_timestamp_micros() -> i64 {
    chrono::Utc::now().timestamp_micros()
}

pub fn now() -> i64 {
    get_now_timestamp_millis()
}

pub fn current_month() -> String {
    chrono::Utc::now().format("%Y-%m").to_string()
}

pub fn add_one_month(timestamp_millis: i64) -> Option<i64> {
    let dt = chrono::DateTime::from_timestamp_millis(timestamp_millis)?;
    let next = dt.checked_add_months(chrono::Months::new(1))?;
    Some(next.timestamp_millis())
}

pub async fn sleep(_duration: std::time::Duration) {
    #[cfg(feature = "web")]
    gloo_timers::future::sleep(_duration).await;

    #[cfg(feature = "server")]
    tokio::time::sleep(_duration).await;
}

pub fn time_ago(timestamp_millis: i64) -> String {
    let now = chrono::Utc::now().timestamp_millis();
    let diff = now - timestamp_millis;

    if diff < 60 * 1000 {
        format!("{}s ago", diff / 1000)
    } else if diff < 3600 * 1000 {
        format!("{}m ago", diff / 1000 / 60)
    } else if diff < 86400 * 1000 {
        format!("{}h ago", diff / 1000 / 3600)
    } else if diff < 604800 * 1000 {
        format!("{}d ago", diff / 1000 / 86400)
    } else if diff < 31536000 * 1000 {
        format!("{}w ago", diff / 1000 / 604800)
    } else {
        format!("{}y ago", diff / 1000 / 31536000)
    }
}
